packages/svelte/src/lib/utils.svelte.ts

```diff
@@ -0,0 +1,64 @@
+import { hasContext, getContext, setContext, untrack } from "svelte";
+import { SvelteMap } from "svelte/reactivity";
+
+export function createContext<T>(name: string) {
+  const key = Symbol(name);
+  return {
+    hasContext: () => {
+      try {
+        return hasContext(key);
+      } catch (e) {
+        if (
+          typeof e === "object" &&
+          e !== null &&
+          "message" in e &&
+          typeof e.message === "string" &&
+          e.message?.includes("lifecycle_outside_component")
+        ) {
+          return false;
+        }
+
+        throw e;
+      }
+    },
+    getContext: () => getContext<T>(key),
+    setContext: (value: T) => setContext(key, value),
+  };
+}
+
+export function promiseWithResolvers<T>(): {
+  promise: Promise<T>;
+  resolve: (value: T) => void;
+  reject: (reason?: unknown) => void;
+} {
+  let resolve: (value: T) => void;
+  let reject: (reason?: unknown) => void;
+  const promise = new Promise<T>((res, rej) => {
+    resolve = res;
+    reject = rej;
+  });
+  return { promise, resolve: resolve!, reject: reject! };
+}
+
+export class KeyedStore<T> extends SvelteMap<string, T> {
```

Comment:
This is a `Map<string, T>` that auto-populates values when getting a key with no existing value, meaning the return type of the `get` function can be `T` instead of `T | undefined` -- v convenient